src/ResourceManager/IotHub/Commands.IotHub/IotHub/SetAzureRmIotHubVerifiedCertificate.cs

```diff
@@ -0,0 +1,169 @@
+
+    [Cmdlet(VerbsCommon.Set, "AzureRmIotHubVerifiedCertificate", DefaultParameterSetName = InputObjectParameterSet, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSCertificateDescription))]
+    [Alias("Set-AzureRmIotHubVC")]
+    public class SetAzureRmIotHubVerifiedCertificate : IotHubBaseCmdlet
+    {
+        private const string ResourceIdParameterSet = "ResourceIdSet";
+        private const string ResourceParameterSet = "ResourceSet";
+        private const string InputObjectParameterSet = "InputObjectSet";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Certificate Object")]
+        [ValidateNotNullOrEmpty]
+        public PSCertificateDescription InputObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the Resource Group")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the Iot Hub")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "Name of the Certificate")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string CertificateName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            HelpMessage = "base-64 representation of X509 certificate .cer file or .pem file path.")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "base-64 representation of X509 certificate .cer file or .pem file path.")]
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "base-64 representation of X509 certificate .cer file or .pem file path.")]
+        [ValidateNotNullOrEmpty]
+        public string Path { get; set; }
```

Comment:
Path should be after Etag, as Etag is still part of identifying the object and it doesn't make sense to separate is from the rest of the parameters identifying the object.

Reply:
Done